client/main.lua

```diff
@@ -143,42 +143,40 @@ local function CreateZone(type, number)
             }
         })
     else
-        CreateThread(function()
-            local function enterZone()
-                if type == 'main' then
-                    TriggerEvent('qb-truckerjob:client:PaySlip')
-                elseif type == 'vehicle' then
-                    TriggerEvent('qb-truckerjob:client:Vehicle')
-                    markerLocation = coords
-                    ShowMarker(true)
-                elseif type == 'stores' then
-                    markerLocation = coords
-                    lib.notify({ title = 'Store Reached', description = Lang:t("mission.store_reached"), duration = 5000, type = 'inform' })
-                    ShowMarker(true)
-                    SetDelivering(true)
-                end
-            end
-            local function exitZone()
-                if type == 'vehicle' then
-                    ShowMarker(false)
-                elseif type == 'stores' then
-                    ShowMarker(false)
-                    SetDelivering(false)
-                end
+        local function enterZone()
+            if type == 'main' then
+                TriggerEvent('qb-truckerjob:client:PaySlip')
+            elseif type == 'vehicle' then
+                TriggerEvent('qb-truckerjob:client:Vehicle')
+                markerLocation = coords
+                ShowMarker(true)
+            elseif type == 'stores' then
+                markerLocation = coords
+                lib.notify({ title = 'Store Reached', description = Lang:t("mission.store_reached"), duration = 5000, type = 'inform' })
+                ShowMarker(true)
+                SetDelivering(true)
             end
-            local boxZones = lib.zones.box({
-                name = boxName,
-                coords = coords,
-                size = size,
-                rotation = rotation,
-                debug = debug,
-                onEnter = enterZone,
-                onExit = exitZone
-            })
-            if type == 'stores' then
-                CurrentLocation.zoneCombo = boxZones
+        end
+        local function exitZone()
+            if type == 'vehicle' then
+                ShowMarker(false)
+            elseif type == 'stores' then
+                ShowMarker(false)
+                SetDelivering(false)
```

Comment:
Can one even have functions within functions like this? If needing to move them out of the OpenMenuGarage() function consider renaming to be more descriptive.

Reply:
Indeed, this is valid Lua. Functions within functions are a thing, as are variables within functions.

Reply:
And no need to move them out since they are only used here.

Reply:
Hmm.. Even so I would go out on a limb and say this should never be done as it seems likely to lead to flaky code. These should probably be moved as anonymous functions within the box function or extracted out altogether.

Reply:
Agreed, should be anonymous functions

Reply:
I'll move them as anonymous functions within the box function, test, then post my changes